### PR TITLE
Don't have requestAnimationFrame looping in background for Live Mode 

### PR DESCRIPTION
**Pull request by eoghanmurray**
Don't have requestAnimationFrame looping in background for Live Mode - instead restart the loop when a new action is added

### CODE REVIEW

File: packages/rrweb/src/replay/timer.ts [hunk 1]
```diff
@@ -10,64 +10,71 @@ export class Timer {
   public speed: number;
 
   private actions: actionWithDelay[];
-  private raf: number | null = null;
-  private liveMode: boolean;
+  private raf: number | boolean | null = null;
```
Comment (Mark-Fenng) [inline packages/rrweb/src/replay/timer.ts:13]:
```suggestion
  private raf: number | true | null = null;
```
I think we can make this type more precise.

File: packages/rrweb/src/replay/index.ts [hunk 2]
```diff
@@ -719,18 +718,16 @@ export class Replayer {
...
-          // defer finish event if the last event is a mouse move
-          setTimeout(() => {
-            finish();
-          }, Math.max(0, -event.data.positions[0].timeOffset + 50)); // Add 50 to make sure the timer would check the last mousemove event. Otherwise, the timer may be stopped by the service before checking the last event.
-        } else {
-          finish();
+          // extend finish event if the last event is a mouse move so that the timer isn't stopped by the service before checking the last event
+          finish_buffer += Math.max(0, -event.data.positions[0].timeOffset);
         }
+        setTimeout(finish, finish_buffer);
```
Comment (Mark-Fenng) [inline packages/rrweb/src/replay/index.ts:730]: With #1115, there will not be 'finish' event anymore in the liveMode. So IMO adding delay for all events won't be necessary.

**Review by Mark-Fenng — COMMENTED** [review]
Most of code looks good to me.

File: packages/rrweb/src/replay/timer.ts [hunk 3]
```diff
@@ -10,64 +10,71 @@ export class Timer {
...
+      this.raf = requestAnimationFrame(this.rafCheck.bind(this));
+    } else {
+      this.raf = true; // was active
+    }
   }
 
   public clear() {
     if (this.raf) {
-      cancelAnimationFrame(this.raf);
+      if (this.raf !== true) {
```
Comment (Juice10) [inline packages/rrweb/src/replay/timer.ts:75]: `raf` can also be null
```suggestion
      if (typeof this.raf === "number") {
```

**Review by Juice10 — APPROVED** [review]

File: packages/rrweb/src/replay/index.ts (hunk 2 above)
Comment (eoghanmurray) [reply packages/rrweb/src/replay/index.ts:730]: Was the removal of this setTimeout implemented in #1115?

**Pull request merged**